src/server/status_server/mod.rs

```diff
@@ -1327,6 +1372,61 @@ mod tests {
         status_server.stop();
     }
 
+    #[test]
+    fn test_metrics() {
+        let _test_guard = TEST_PROFILE_MUTEX.lock().unwrap();
+        let mut status_server = StatusServer::new(
+            1,
+            ConfigController::default(),
+            Arc::new(SecurityConfig::default()),
+            MockRouter,
+            std::env::temp_dir(),
```

Comment:
Should use `tempfile::TempDir` instead, otherwise they won't be cleaned up.

Reply:
done. Replaced all with `tempfile::TempDir`